src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -212,12 +212,18 @@ private void OnActiveDocumentChanged(object? sender, DocumentId? activeDocumentI
                     return;
 
                 RoslynDebug.AssertNotNull(activeDocumentId);
+
+                // When the analysis scope is set to 'ActiveFile' and the active document changes,
+                // analysis results might change for the analyzers whose document analysis results depends on
+                // whether or not the document is active. We trigger reanalysis for the active document for these analyzers.
                 var analysisScope = SolutionCrawlerOptions.GetBackgroundAnalysisScope(activeProject);
                 if (analysisScope == BackgroundAnalysisScope.ActiveFile)
                 {
-                    // When the active document changes and we are only analyzing the active file, trigger a document
-                    // changed event to reanalyze the newly-active file.
-                    EnqueueEvent(solution, activeDocumentId, InvocationReasons.DocumentChanged, nameof(OnActiveDocumentChanged));
```

Comment:
💭 It seems like it would be much easier to just have a new `InvocationReasons.ActiveDocumentChanged`, and let the one analyzer that needs it handle the analysis scope there.

Reply:
i really like this idea.

Reply:
I have implemented this suggestion. I have named it as `InvocationReasons.ActiveDocumentSwitched` instead of `InvocationReasons.ActiveDocumentChanged` as this might confuse one as to whether the text for the active document changed, while the event really just signifies active tab change.